### PR TITLE
chore: Removes mentions and examples of Serverless and Shared-tier instances

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): CLOUDP-284805

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: docs/resources/cluster.md [hunk 1]
```diff
@@ -187,19 +187,6 @@ resource "mongodbatlas_cluster" "cluster-test" {
   }
 }
 ```
-### Example AWS Shared Tier (M2/M5) cluster
```
Comment (Zuhairahmed) [inline docs/resources/cluster.md:190]: on Advanced Cluster page Tenant example https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster#example-tenant-cluster can we also update example from "M5" to "M0"? all good from my end after you make this update

**Review by Zuhairahmed — APPROVED** [review]
left one comment, feel free to merge after updating

File: docs/resources/cluster.md (hunk 1 above)
Comment (oarbusi) [reply docs/resources/cluster.md:190]: Done!

File: templates/data-source.md.tmpl [hunk 2]
```diff
@@ -9,9 +9,7 @@
     {{ else if eq .Name "mongodbatlas_privatelink_endpoint" }}
         {{ tffile (printf "examples/%s/aws/cluster/main.tf" .Name )}}
     {{ else if eq .Name "mongodbatlas_privatelink_endpoint_service_serverless" }}
```
Comment (lantoli) [inline templates/data-source.md.tmpl:11]: `else if` can also be deleted as they're empty now for deleted if content

**Review by lantoli — APPROVED** [review]

**Review by carriecwk — APPROVED** [review]
lgtm

**Review by cveticm — APPROVED** [review]

**Pull request merged**